src/flamenco/runtime/fd_executor.c

```diff
@@ -1266,6 +1267,14 @@ fd_executor_setup_borrowed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
           || ( i>0UL && fd_should_set_exempt_rent_epoch_max( txn_ctx->slot_ctx, borrowed_account ) ) ) {
         borrowed_account->meta->info.rent_epoch = ULONG_MAX;
       }
+    } else {
+      /* When direct mapping is enabled, the account data has to be 16-byte aligned, otherwise memory translation inside the vm
+         will fail non-deterministically. */
+      if ( FD_UNLIKELY( FD_FEATURE_ACTIVE( txn_ctx->slot_ctx, bpf_account_data_direct_mapping ) ) ) {
+        uchar * aligned_const_data = fd_valloc_malloc( txn_ctx->valloc, FD_VM_HOST_REGION_ALIGN, borrowed_account->const_meta->dlen );
+        fd_memcpy( aligned_const_data, borrowed_account->const_data, borrowed_account->const_meta->dlen );
+        borrowed_account->const_data = aligned_const_data;
+      }
```

Comment:
instead of always copying, couldn't we check to see if const_data is already aligned and only copy if it is not?  Or, is it never aligned 16 byte aligned due to the size of the header?  This feels like we should either tweak the header or hack funk.  memcpy's are already a large portion of our execution time